java/arcs/core/data/Recipe.kt

```diff
@@ -23,7 +23,7 @@ data class Recipe(
 ) {
     /** The name of the policy with which this recipe must comply. */
     val policyName: String?
-        get() = annotations.find { it.name === "policy" }?.getStringParam("name")
```

Comment:
Ah, this was not caught because there were no tests for this?

Reply:
There actually is a test for this already in `RecipeTest.kt`. It's just that it passes using `===` reference equality... The test looks like this:

```
        assertThat(
            Recipe(
                name = "Foo",
                handles = emptyMap(),
                particles = emptyList(),
                annotations = listOf(Annotation.createPolicy("MyPolicy"))
            ).policyName
        ).isEqualTo("MyPolicy")
```

I think it's because the `"MyPolicy"` string gets statically allocated once, and so both references to it are the same. Whereas when we decode the policy name from a proto, it is a separate allocation and so `===` doesn't work.

...At least, that's what I suspect. I didn't dig into it beyond that.

Reply:
What you say makes sense. We were just lucky before! :)